src/oncall/auth/__init__.py

```diff
@@ -186,6 +187,12 @@ def authenticate_application(auth_token, req):
 
 
 def _authenticate_user(req):
+    global sso_auth_manager
+    # pass the req to the sso_auth_manager so it can check if it has valid SSO headers
+    if sso_auth_manager and sso_auth_manager.authenticate(req):
```

Comment:
Looks like this code calls "authenticate()" twice, which might be expensive. 

Can it be changed to only call it once? The walrus operator could be helpful here to avoid some boilerplate.

Reply:
good point, there's really no reason to call this twice. Changed